File: 0066-plus-one/66.plus-one.py
#
# @lc app=leetcode id=66 lang=python3
#
# [66] Plus One
#

# @lc code=start
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        i = 1
        while i <= len(digits):
            if digits[-i] != 9:
                digits[-i] += 1
                return digits
                
            digits[-i] = 0
            i += 1

        return [1] + digits
# @lc code=end

File: 0066-plus-one/66.plus-one.cs
/*
 * @lc app=leetcode id=66 lang=csharp
 *
 * [66] Plus One
 */

// @lc code=start
public class Solution {
    public int[] PlusOne(int[] digits) {
        for (int i = digits.Length - 1; i >= 0; i--)
        {
            if (digits[i] != 9)
            {
                digits[i]++;
                return digits;
            }
            digits[i] = 0;
        }

        int[] newDigit = new int[digits.Length+1];
        newDigit[0] = 1;

        return newDigit;
    }
}
// @lc code=end

